src/main/java/net/hockeyapp/android/CrashManager.java

```diff
@@ -324,22 +346,68 @@ public static void deleteStackTraces(WeakReference<Context> weakContext) {
           if (weakContext != null) {
             Log.d(Constants.TAG, "Delete stacktrace " + list[index] + ".");
             deleteStackTrace(weakContext, list[index]);
-            
+
             context = weakContext.get();
             if (context != null) {
-              context.deleteFile(list[index]);              
+              context.deleteFile(list[index]);
             }
           }
-        } 
+        }
         catch (Exception e) {
           e.printStackTrace();
         }
       }
     }
   }
-  
+
+  /**
+   Provides an interface to pass user input from a custom alert to a crash report
+
+   @param userInput Defines the users action wether to send, always send, or not to send the crash report.
+   @param userProvidedMetaData The content of this optional BITCrashMetaData instance will be attached to the crash report and allows to ask the user for e.g. additional comments or info.
+
+   @return Returns YES if the input is a valid option and successfully triggered further processing of the crash report
+
+   @see CrashManagerUserInput
+   @see CrashMetaData
+   */
+  public static boolean handleUserInput(final CrashManagerUserInput userInput,
+      final CrashMetaData userProvidedMetaData, final CrashManagerListener listener,
+      final WeakReference<Context> weakContext, final boolean ignoreDefaultHandler) {
+    switch (userInput) {
+      case CrashManagerUserInputDontSend:
+        if (listener != null) {
+          listener.onUserDeniedCrashes();
+        }
+
+        deleteStackTraces(weakContext);
+        registerHandler(weakContext, listener, ignoreDefaultHandler);
+        return true;
+      case CrashManagerUserInputSend:
+        Context context = null;
+        if (weakContext != null) {
+          context = weakContext.get();
+        }
+
+        if (context == null) {
+          return false;
+        }
+
+        final SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
```

Comment:
This is in the wrong spot, isn't it? Should be in `case CrashManagerUserInputAlwaysSend:`, right?